eu_einvoice/european_e_invoice/custom/sales_invoice.py

```diff
@@ -744,10 +744,18 @@ def validate_doc(doc, event):
 			indicator="orange",
 		)
 
-	validate_einvoice(doc)
+	validate_einvoice(doc, event)
 
 
-def validate_einvoice(doc: SalesInvoice):
+def validate_einvoice(doc: SalesInvoice, event: str):
+	if event == "before_save":
+		validation_state = frappe.get_single_value("E Invoice Settings", "validation_on_save_insert")
+	if event == "before_submit":
+		validation_state = frappe.get_single_value("E Invoice Settings", "validation_on_submit")
```

Comment:
```suggestion
	if event not in ["before_save", "before_submit"]:
		return
	settings_field = "validation_on_save_insert" if event == "before_save" else "validation_on_submit"
	validation_state = frappe.get_single_value("E Invoice Settings", settings_field)
```
Little refactoring (just suggestion).